Firestore/Example/FuzzTests/FSTFuzzTestsPrincipal.mm

```diff
@@ -49,27 +59,83 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
   return 0;
 }
 
+// Retrieves the fuzzing target from the FUZZING_TARGET environment variable.
+// Default target is NONE if the environment variable is empty, not set, or
+// could not be interpreted. Should be kept in sync with FuzzingTarget.
+FuzzingTarget GetFuzzingTarget() {
+  NSString *fuzzing_target_env =
+      [[[NSProcessInfo processInfo] environment] objectForKey:@"FUZZING_TARGET"];
+
+  if (fuzzing_target_env == nil || [fuzzing_target_env length] == 0) {
+    LOG_WARN("No value provided for FUZZING_TARGET environment variable.");
+    return NONE;
+  }
+
+  if ([@"NONE" isEqualToString:fuzzing_target_env]) {
+    return NONE;
+  }
+
+  if ([@"SERIALIZER" isEqualToString:fuzzing_target_env]) {
+    return SERIALIZER;
+  }
+
+  // Value did not match any target.
+  LOG_WARN("Invalid fuzzing target: %s", fuzzing_target_env);
+
+  // Default is NONE.
+  return NONE;
+}
+
 // Simulates calling the main() function of libFuzzer (FuzzerMain.cpp).
+// Uses GetFuzzingTarget() to get the fuzzing target and sets libFuzzer's args
+// accordingly.
 int RunFuzzTestingMain() {
-  // Get dictionary file path from resources and convert to a program argument.
+  // Get the fuzzing target.
+  FuzzingTarget fuzzing_target = GetFuzzingTarget();
+
+  // All fuzzing resources.
+  NSString *resources_location = @"Firestore_FuzzTests_iOS.xctest/FuzzingResources";
+
+  // The dictionary location for the fuzzing target.
+  NSString *dict_location;
+
+  // The corpus location for the fuzzing target.
+  NSString *corpus_location;
+
+  // Set the dictionary and corpus locations according to the fuzzing target.
+  switch (fuzzing_target) {
+    case SERIALIZER:
+      dict_location =
+          [resources_location stringByAppendingPathComponent:@"Serializer/serializer.dictionary"];
+      corpus_location = @"FuzzTestsCorpus";
+      break;
+
+    case NONE:
+    default:
+      LOG_WARN("Not going to run fuzzing, exiting!");
+      return 0;
+  }
+
+  // Get dictionary and corpus paths from resources and convert to program arguments.
   NSString *plugins_path = [[NSBundle mainBundle] builtInPlugInsPath];
 
-  NSString *dict_location =
-      @"Firestore_FuzzTests_iOS.xctest/FuzzingResources/Serializer/serializer.dictionary";
   NSString *dict_path = [plugins_path stringByAppendingPathComponent:dict_location];
   const char *dict_arg = [[NSString stringWithFormat:@"-dict=%@", dict_path] UTF8String];
 
-  // Get corpus and convert to a program argument.
-  NSString *corpus_location = @"FuzzTestsCorpus";
   NSString *corpus_path = [plugins_path stringByAppendingPathComponent:corpus_location];
   const char *corpus_arg = [corpus_path UTF8String];
 
+  // The directory in which libFuzzer writes crashing inputs.
+  const char *prefix_arg =
+      [[@"-artifact_prefix=" stringByAppendingString:kCrashingInputsDirectory] UTF8String];
```

Comment:
I might be missing something about how Objective-C works, but it looks like a dangling pointer (the temporary created by `stringByAppendingString` will have its lifetime ended right after this line, but `prefix_arg` still points to a buffer owner by it). How about:
```cpp
#include "Firestore/core/src/firebase/firestore/util/string_apple.h"

using firebase::firestore::util::MakeString;

//...

std::string prefix_arg = std::string("-artifact_prefix=") + MakeString(kCrashingInputsDirectory);
// ...
const_cast<char*>(prefix_arg.c_str());
```
(or use initialization similar to `corpus_arg` and other arguments above. Since it stores the created string in a named variable, lifetime is not an issue)

Reply:
I changed everything to use the `std::string + MakeString`